velox/common/memory/MemoryPool.cpp

```diff
@@ -491,7 +491,9 @@ void* MemoryPoolImpl::allocate(
     std::optional<uint32_t> alignment) {
   if (alignment.has_value()) {
     const auto alignmentValue = alignment.value();
-    if (FOLLY_UNLIKELY(alignment_ % alignmentValue != 0)) {
+    if (FOLLY_UNLIKELY(
+            !(MemoryAllocator::isPowerOfTwo(alignmentValue) &&
+              alignmentValue <= alignment_))) {
```

Comment:
ditto BitUtil.h

And I think you mean to put || condition?

Reply:
It's &&, alignment is only allowed to be power of two, and the alignment of allocation must be guaranteed by alignment of the pool

Reply:
Right, I missed a bracket, thinking the ! is applied to only the first condition.